packages/application-shell-connectors/src/components/application-state/README.md

```diff
@@ -0,0 +1,31 @@
+# Connectors for `applicationState`
+
+This component exposes getters/setters to set and retrieve the `applicationState`.
+
+> It uses the new React Context API, so you need to provide the `ApplicationStateProvider` up in the tree. This is done within the AppShell!
+
+Additionally it provides a HOC to inject the `applicationState`.
+
+Use this component to access about the `user`, `project` and application `environment`.
+
+### Usage
+
+```js
+/* In the AppShell */
+<ApplicationStateProvider
+  user={{...}}
+  project={{...}}
+  projectDataLocale="en"
+  environment={{...}}
+>
+  <div>
+    {/* ... */}
+    {/* In the application specific code */}
+    <GetApplicationState
```

Comment:
I personally like more the `with` for HOCs (because we are **injecting** something) and `Get` for components, because it's a getter after all.

But yeah, it's a subtle difference. If it's important to settle on a name, we should then write an ADR or something for future implementations. WDYT?

Reply:
I don't think it's crucial. Just wanted to mention it. Various libraries go different about it anyway: `injectIntl`, `injectFeatureToggle`, `withRouter`. I think we can agree that `with*` is good for the HoC. `injectIntl`, if I recall correctly, renders a `WithIntl`. I am fine with the `Get*` just haven't seen it anywhere else (as in other libraries) and was curious as to why we deviate.

Reply:
Well yeah, I would say it's kind of our own convention. I think as long as it makes sense and it's clear for the consumers on what it does, it's fine to use it.

Also, those `inject` functions have been there for a long time, before all those new patterns started to come out (e.g. render props, provider/consumer, etc).

But again, this is just my personal taste, I'm fine also if we agree on something more "standard".